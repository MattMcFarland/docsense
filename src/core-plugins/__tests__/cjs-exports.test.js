const testFactory = require('./utils/testFactory')
const cjsExportsPlugin = require('../cjs-exports')
const defaultExports = [
  'module.exports = function () {}',
  'exports = function () {}',
  'module.exports = foo()',
  'exports = foo()',
  'module.exports = new Foo()',
  'exports = new Foo()',
  'module.exports = new a.b.c.Foo()',
  'exports = new a.b.c.Foo()',
  'module.exports = []',
  'exports = []',
  'module.exports = () => {}',
  'exports = () => {}',
  'module.exports = fn.call()',
  'exports = fn.call()',
  'module.exports = fn.apply()',
  'exports = fn.apply()',
  '(function () { module.exports = foo })()',
]
const namedExports = [
  'module.exports.name = function () {}',
  'exports.name = function () {}',
  'module.exports.name = foo()',
  'exports.name = foo()',
  'module.exports.name = []',
  'exports.name = []',
  'module.exports.name = () => {}',
  'exports.name = () => {}',
  'module.exports.name = fn.call()',
  'exports.name = fn.call()',
  'module.exports.name = fn.apply()',
  'exports.name = fn.apply()',
]
const objectExports = [
  'module.exports = { name1, name2, name3 }',
  'exports = { name1, name2, name3 }',
]
const notExports = [
  'var exports = 4',
  'let foo = { exports: 4}',
  'const foo = bar.exports',
  'const foo = derp.exports',
  'const foo = bar.exports.baz',
  'foo.map(exports => exports.module)',
  'foo.map(module => module.exports = bar)',
  'module.other.exports = bar',
  'module.other.exports = { exports }',
  'module.other.exports.foo = bar',
  'module.other.exports.foo = { exports }',
  'some.module.exports = baz',
  'some.module.exports = { exports }',
  'some.exports = baz',
  'some.module.exports.foo = baz',
  'some.module.exports.foo = { exports }',
  'some.exports.foo = baz',
  'some.exports.foo = { exports }',
  'some.module.other.exports = bar',
  'some.module.other.exports = { exports }',
  'some.other.exports = bar',
  'some.other.exports = { exports }',
  'some.module.other.exports.foo = baz',
  'some.module.other.exports.foo = { exports }',
  'some.other.exports.foo = baz',
  'some.other.exports.foo = { exports }',
]
const suites = {
  defaultExports,
  namedExports,
  objectExports,
  notExports,
}
testFactory({
  name: 'Core Plugin: Common.js Exports',
  plugin: cjsExportsPlugin,
  suites,
})
