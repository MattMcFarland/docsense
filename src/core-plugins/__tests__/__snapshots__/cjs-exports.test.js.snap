// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core Plugin: Common.js Exports defaultExports (function () { module.exports = foo })() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = new Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports exports = new a.b.c.Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = new Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports defaultExports module.exports = new a.b.c.Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports exports.name = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports exports.name = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports exports.name = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports exports.name = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports exports.name = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports exports.name = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports module.exports.name = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports module.exports.name = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports module.exports.name = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports module.exports.name = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports module.exports.name = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports namedExports module.exports.name = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports notExports const foo = bar.exports 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports const foo = bar.exports.baz 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports const foo = derp.exports 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports foo.map(exports => exports.module) 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports foo.map(module => module.exports = bar) 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports let foo = { exports: 4} 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports module.other.exports = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports module.other.exports = bar 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports module.other.exports.foo = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports module.other.exports.foo = bar 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.exports = baz 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.exports.foo = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.exports.foo = baz 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.exports = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.exports = baz 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.exports.foo = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.exports.foo = baz 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.other.exports = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.other.exports = bar 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.other.exports.foo = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.module.other.exports.foo = baz 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.other.exports = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.other.exports = bar 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.other.exports.foo = { exports } 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports some.other.exports.foo = baz 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports notExports var exports = 4 1`] = `Array []`;

exports[`Core Plugin: Common.js Exports objectExports exports = { name1, name2, name3 } 1`] = `
Array [
  Object {
    "cjsExports_id": "name1",
    "file_id": "__TEST__",
  },
  Object {
    "cjsExports_id": "name2",
    "file_id": "__TEST__",
  },
  Object {
    "cjsExports_id": "name3",
    "file_id": "__TEST__",
  },
]
`;

exports[`Core Plugin: Common.js Exports objectExports module.exports = { name1, name2, name3 } 1`] = `
Array [
  Object {
    "cjsExports_id": "name1",
    "file_id": "__TEST__",
  },
  Object {
    "cjsExports_id": "name2",
    "file_id": "__TEST__",
  },
  Object {
    "cjsExports_id": "name3",
    "file_id": "__TEST__",
  },
]
`;
